content/challenges/hard/2 - Recurse Palindrome/func.js

```diff
@@ -10,5 +10,16 @@
  * @returns {boolean} true if the string is a palindrome, false otherwise
  */
 module.exports = (str) => {
-  // your code here
+  function symmetryCheck(lettersArray) {
+    if (lettersArray.length <= 1) {
+      return true;
+    }
+    if (lettersArray.shift() === lettersArray.pop()) {
+      return symmetryCheck(lettersArray);
+    } else {
+      return false;
+    }
+  }
+
+  return symmetryCheck(str.split(''));
```

Comment:
Interesting way to keep the algorithm 0(n/2). i see you also used recursion. 

now...can you do this one without using `shift()` or `pop()` and just use primitive methods to solve this?